de-lib/src/main/java/org/iplantc/de/tools/client/presenter/ManageToolsViewPresenter.java

```diff
@@ -347,35 +325,27 @@ public void onSuccess(NewToolRequestDialog o) {
 
     @Override
     public void onEditToolSelected(EditToolSelected event) {
-        editDialogProvider.get(new AsyncCallback<EditToolDialog>() {
+        toolServices.getToolInfo(getSelectedTool().getId(), new AppsCallback<Tool>() {
             @Override
-            public void onFailure(Throwable throwable) {
-
+            public void onFailure(Integer statusCode, Throwable exception) {
+                announcer.schedule(new ErrorAnnouncementConfig(appearance.toolInfoError()));
             }
 
             @Override
-            public void onSuccess(EditToolDialog etd) {
-                etd.setSize(appearance.editDialogWidth(), appearance.editDialogHeight());
-                etd.show(ManageToolsViewPresenter.this);
-                etd.mask();
```

Comment:
Ha!  I did it this way because I saw you did it this way 😆 

So the issues I ran into when I initially did it the other way is `enableReinitialize` doesn't seem to work perfectly.  If I initialize with empty values (or really, initialize with the `DEFAULT_TOOL` value), then reinitialize with real values filled in, the `TextFields` look like this:
![image](https://user-images.githubusercontent.com/8909156/61165105-eeea5c00-a4d0-11e9-840d-0f23024078dc.png)

The helper text overlaps with the value because I guess it doesn't recognize that a value is actually there now.

If I populate `DEFAULT_TOOL` with every possible key/value pair (right now I'm only populating the required fields so the validation checks will work), I believe it works, but then I send empty strings to the service which I thought we didn't want.

Reply:
🤔 I agree, I don't think we want to send the empty strings to the service.
I ran into issues in the `EditMetadata` form with resetting these values too (I can't remember the specifics now), so I ended up not setting `enableReinitialize`, and instead I manually call `resetForm` if metadata for a new file/folder is getting edited, but also when the metadata initially loads from the service:
https://github.com/cyverse-de/ui/blob/686fe142156774e15aa76e0581944715843aa9cf/react-components/src/metadata/EditMetadata.js#L86-L96
and
https://github.com/cyverse-de/ui/blob/686fe142156774e15aa76e0581944715843aa9cf/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/DiskResourceMetadataViewImpl.java#L55-L61

I had to track the `targetResource` separate from the `metadata` since all the `metadata` could be updated by a template, but maybe a simpler check could be made with the `tool` in this form?

I'm not sure if this will help with that odd display of the `TextFields` after loading the tool, though.

Reply:
I just tested it with `resetForm` and it doesn't seem to fix the issue with `TextFields`.  It does seem to be a known issue with material-ui, so perhaps not a formik issue: https://material-ui.com/components/text-fields/#limitations

I realized with JavaScript it actually wouldn't be that hard to write a function to clear out the empty values, so I could go that route instead and fully populate `DEFAULT_TOOL` with all possible key value pairs:

```
function clearEmptyValues(values) {
    let filteredValues = {};

    Object.keys(values).forEach(key => {
        if (values[key] && typeof values[key] === "object") {
            filteredValues[key] = clearEmptyValues(values[key]);
        } else if (values[key] != null && values[key] !== "") {
            filteredValues[key] = values[key];
        }
    });

    return filteredValues;
}
```

Reply:
Ah, one snag with that though is if you fill in, for example, a description and then want to remove that description it doesn't work anymore.  Not sure how common that is though.

Reply:
Maybe we could have a generic LoadingDialog that gets loaded instead?  Similar to this idea, sort of: https://cyverse-ui.surge.sh/components/skeleton-list-doc

Reply:
That might work. We definitely want to support allowing a field to be cleared.

Another option would be to put a loading mask on the "Manage Tools" listing instead, until the dialog is ready to display.

Reply:
Ah, simple solution!  I'll go that route.